### DISCUSSION TITLE
XMPP: Post to PubSub

### DISCUSSION THREAD

**Discussion by sjehuda** [joomla/joomla-cms#43327] [2. Cold Features]
### Is your feature request related to a problem? Please describe.

I want to publish my Backdrop contents over PubSub node items on my XMPP server.

### Describe the solution you'd like

Add support to post to pubsub node items using Moxl library.

### Additional context

Greetings!

In XMPP, using PubSub(1) (XEP-0060), it is possible to store posts as Atom Syndication(2) entries into XMPP(3).

There are several XMPP extensions that apply it, namely XEP-0227 (4) and XEP-0472 (5) which make node `urn:xmpp:microblog:0` as a common choice to manage news for client accounts (e.g. `blog@joomla.org`), yet a component account (e.g. `pubsub.joomla.org`) may use all of its nodes for this purpose.

```
pubsub.joomla.org/
|
|-- updates/
    |-- Sombre
     `-- Varied
|-- releases/
 `-- events/
```
```
pubsub service/
|
|-- node/
    |-- item
     `-- item
|-- node/
 `-- node/
```

A PHP library that can be utilized would be Moxl of Movim.

I have made a script that realizes viewing PubSub node items as Atom XML feeds over HTTP.

1. [XEP-0060: Publish-Subscribe](https://xmpp.org/extensions/xep-0060.html)
2. [RFC 4287: The Atom Syndication Format](https://www.rfc-editor.org/rfc/rfc4287)
3. [Atom Over XMPP](https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf)
4. [XEP-0277: Microblogging over XMPP](https://xmpp.org/extensions/xep-0277.html)
5. [XEP-0472: Pubsub Social Feed](https://xmpp.org/extensions/xep-0472.html)
6. [Moxl](https://github.com/movim/movim/tree/master/src/Moxl)
7. [PubSub Over HTTP](https://codeberg.org/poezio/slixmpp/issues/3526)

**Comment by rdeutz** [comment]
This needs better desciption what are the advantages and use cases for this. I not 100% against this but I don't see that we will working on this in the nearer future nor that this makes sense at all.

**Reply by sjehuda** [reply]
Robert, thank you for accepting this to discussion!

I have elaborated on it at https://github.com/xenocrat/chyrp-lite/issues/250#issuecomment-2066270209

One of the advantages I presented there, beyond an additional publishing platform, is creating a direct rapport with people which you chat with on the XMPP platform; while you chat, they might decide to check your recent posts and even make comments on them, either in the PubSub node or directly comment to you about it in chat.

Please tell me if the explanation to which I have linked to is not suffice.

**Comment by Neustradamus** [comment]
To follow this discussion, an important feature request!
